src/frontend/src/components/tables/Search.tsx

```diff
@@ -0,0 +1,36 @@
+import { CloseButton, TextInput } from '@mantine/core';
+import { IconSearch } from '@tabler/icons-react';
+import { useEffect, useState } from 'react';
+
+export function TableSearchInput({
+  searchCallback
+}: {
+  searchCallback: (searchTerm: string) => void;
+}) {
+  const [value, setValue] = useState<string>('');
+  const [searchValue, setSearchValue] = useState<string>('');
+
+  useEffect(() => {
+    const timer = setTimeout(() => {
+      if (value != searchValue) {
+        setSearchValue(value);
+        searchCallback(value);
+      }
+    }, 500);
+    return () => clearTimeout(timer);
+  });
```

Comment:
This denounce logic could be replaced by a useDebounce hook from [mantine](https://mantine.dev/hooks/use-debounced-value/).